imdown-mobile/events/event_details.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { useState } from "react";
+import { FlatList, Image, Linking, Text, TouchableOpacity, View } from "react-native";
+import { EventDetailsStyles } from "./event_details_styles";
+import moment from 'moment';
+import Divider  from '../components/divider/divider'
+
+const DOWN_EMOJI_HEIGHT = 82
+const DOWN_EMOJI_WIDTH = 85
+const EVENT_PIC_HEIGHT = 130
+const EVENT_PIC_WIDTH = 130
+const ROW_BUTTON_HEIGHT = 40
+const ROW_BUTTON_WIDTH = 40
+
+const EventDetails = (props) => {
+const [event, setEvent] = useState(props.route.params.event)
+
+	const renderRSVPUser = ({ item }: { item: String }) => {
+    return (
+      <View>
+            <Text style={EventDetailsStyles.rsvp_user}>{item}</Text>
+      </View>
+    );
+	};
+
+	const renderBottomRowButtons = () => {
```

Comment:
I really like this. Can we do this for the following parts of the page as well:
`{/* Event title + pic + details box */}`
`{/* Event description box */}`
`{/* Event "down list" box */}`

I think this kind of compartmentalizing makes it easier to follow the layout

Reply:
I am a little wary of doing this, because as discovered today, moving logic into functions translates reading from stateful to stateless. We can do this, but not too sure if in the future, if we want stateful interaction, how much effort is needed to pass in stateful params.